File: caesar/python/caesar.py
#!/usr/bin/python3

import sys

ALPHABET: str = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

def rotateChr(char: str, rot: int) -> str:
    plainIndex = ALPHABET.find(char)
    if (plainIndex < 0):
        return char
    rotIndex = (plainIndex+rot)%len(ALPHABET)
    return ALPHABET[rotIndex]

def rotateStr(text: str, rot: int) -> str:
    result = ""
    for char in text:
        result += rotateChr(char, rot)
    return result

def main() -> None:
    rot = int(sys.argv[1])
    for line in sys.stdin:
        print(rotateStr(line, rot).strip())

if __name__ == "__main__":
    main()

File: caesar/html/caesar.js
let ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

function rotateChr(char, rot) {
    if (rot < 0) {
        rot = ALPHABET.length - (-rot % ALPHABET.length);
    }
    plainIndex = ALPHABET.indexOf(char);
    if (plainIndex < 0) {
        return char
    }
    rotIndex = (plainIndex + rot) % ALPHABET.length;
    return ALPHABET[rotIndex];
}

function rotateStr(text, rot) {
    let result = "";
    for (let i = 0; i < text.length; i++) {
        result += rotateChr(text.charAt(i), rot);
    }
    return result
}